src/pasta/fields/fp.rs

```diff
@@ -646,43 +638,43 @@ impl ff::PrimeField for Fp {
 impl FieldExt for Fp {
     const ROOT_OF_UNITY: Self = ROOT_OF_UNITY;
     const ROOT_OF_UNITY_INV: Self = Fp::from_raw([
-        0x9246674078fa45bb,
-        0xd822ebd60888c5ea,
-        0x56d579133a11731f,
-        0x1c88fa9e942120bb,
+        0xf0b87c7db2ce91f6,
+        0x84a0a1d8859f066f,
+        0xb4ed8e647196dad1,
+        0x2cd5282c53116b5c,
     ]);
     const UNROLL_T_EXPONENT: [u64; 4] = [
-        0x3b3a6633d1897d83,
-        0x0000000000c93d5b,
-        0xf000000000000000,
-        0xe34ab16,
+        0x955a0a417453113c,
+        0x0000000022016b89,
+        0xc000000000000000,
+        0x3f7ed4c6,
     ];
     const T_EXPONENT: [u64; 4] = [
-        0xb61facdcd0a03271,
-        0x0000000001c55093,
+        0x094cf91b992d30ed,
+        0x00000000224698fc,
         0x0000000000000000,
-        0x20000000,
+        0x40000000,
     ];
     const DELTA: Self = DELTA;
-    const UNROLL_S_EXPONENT: u64 = 0x11cb54e91;
+    const UNROLL_S_EXPONENT: u64 = 0x204ace5;
     const TWO_INV: Self = Fp::from_raw([
-        0xd0a0327100000001,
-        0x01c55093b61facdc,
+        0xcc96987680000001,
+        0x11234c7e04a67c8d,
         0x0000000000000000,
         0x2000000000000000,
     ]);
     const RESCUE_ALPHA: u64 = 5;
     const RESCUE_INVALPHA: [u64; 4] = [
-        0x810050b4cccccccd,
-        0x360880ec56991494,
+        0xe0f0f3f0cccccccd,
+        0x4e9ee0c9a10a60e2,
         0x3333333333333333,
         0x3333333333333333,
     ];
     const ZETA: Self = Fp::from_raw([
-        0x8598abb3a410c9c8,
-        0x7881fb239ba41a26,
-        0x9bebc9146ef83d9a,
-        0x1508415ab5e97c94,
+        0x1dad5ebdfdfe4ab9,
+        0x1d1f8bd237ad3149,
+        0x2caad5dc57aab1b0,
+        0x12ccca834acdba71,
     ]);
```

Comment:
I did not check these constants, but I am satisfied by their unit tests.

Reply:
Checked up to here (except things I explicitly said I couldn't check).